File: admin/src/lib/api/media.ts
import { api } from "./client";
import type { APIResponse, PaginatedResponse } from "./types";

export interface MediaAsset {
  id: number;
  filename: string;
  original_name: string;
  mime_type: string;
  file_size: number;
  width: number;
  height: number;
  url: string;
  thumbnail_url: string;
  alt_text: string;
  folder: string;
  uploaded_by: number;
  created_at: string;
}

export const mediaAPI = {
  list: (params?: Record<string, string | number>) =>
    api.get<APIResponse<PaginatedResponse<MediaAsset>>>("/media", params),

  get: (id: number | string) =>
    api.get<APIResponse<MediaAsset>>(`/media/${id}`),

  upload: (file: File, folder = "general", altText = "") => {
    const formData = new FormData();
    formData.append("file", file);
    formData.append("folder", folder);
    formData.append("alt_text", altText);
    return api.upload<APIResponse<MediaAsset>>("/media/upload", formData);
  },

  delete: (id: number | string) =>
    api.delete<APIResponse<null>>(`/media/${id}`),
};

File: bolg-api/internal/model/media.go
package model

import "time"

type MediaAsset struct {
	ID           uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Filename     string    `gorm:"type:varchar(255);not null" json:"filename"`
	OriginalName string    `gorm:"type:varchar(255);not null" json:"original_name"`
	MimeType     string    `gorm:"type:varchar(100);index;not null" json:"mime_type"`
	FileSize     int64     `gorm:"not null" json:"file_size"`
	Width        int       `gorm:"default:0" json:"width"`
	Height       int       `gorm:"default:0" json:"height"`
	URL          string    `gorm:"type:varchar(500);not null" json:"url"`
	ThumbnailURL string    `gorm:"type:varchar(500)" json:"thumbnail_url"`
	AltText      string    `gorm:"type:varchar(255)" json:"alt_text"`
	Folder       string    `gorm:"type:varchar(100);index" json:"folder"`
	UploadedBy   *uint     `gorm:"index" json:"uploaded_by"`
	CreatedAt    time.Time `json:"created_at"`
}
